src/Quartz/Core/QuartzSchedulerThread.cs

```diff
@@ -22,9 +22,10 @@
 using System.Data.Common;
 
 using Microsoft.Extensions.Logging;
-
+using Microsoft.VisualStudio.Threading;
```

Comment:
Core Quartz should not have this kind of dependencies

Reply:
Ok, that is the key to what I'm doing here. There are similar tools found in https://github.com/StephenCleary/AsyncEx that we could use instead. I chose to use the Microsoft implementation because I assume they are going to be more popular and better maintained over time. Are you okay with using the Stephen Cleary libraries instead?

Reply:
I'm still undecided. Because both are MIT licensed, we could just extract the minimal needed parts and add to Quartz codebase with attribution in each file's header...

Reply:
Ok, I ended up doing that with the PauseToken anyway.